### ISSUE TITLE
Create proj_geom shows interesting behaviour. 

### ISSUE THREAD

**Issue by MaxBe23** [astra-toolbox/astra-toolbox#301]
Hey, 

for my reconstruction I'm using the following function: 

> proj_geom = 
> astra.create_proj_geom('cone',
> detector_spacing_x,  # – Distance between two adjacent detector pixels. 
> detector_spacing_y, # – Distance between two adjacent detector pixels.
> det_row_counts, # – Number of detector pixel rows.
> det_col_counts, # – Number of detector pixel columns.
> angles, # – Array of angles in radians.
> source_origin, #– Distance between point source and origin.
> origin_det #– Distance between the detector and origin.)

I set all values to mm to be sure that all works. If I'm doing this I get really small images and the reconstructed volume is really small, what makes no sense. 

My parameters: 

detector_spacing_x/y = 0.24mm
det_row_count = 654 
det_col_counts = 664 
angles = 0:0.4:200 (500 Projections)
source_origin = 428.5 mm 
oring_det = 171.5 mm 

Then I used a trick by a tutorial: 

[Link to tutorial](https://tomroelandts.com/articles/astra-toolbox-tutorial-reconstruction-from-projection-images-part-2)

Refered to the tutorial I changed my code to this: 

> proj_geom = astra.create_proj_geom('cone', 
> 1, 
> 1, 
>                                         int(params['nu']),
>                                         int(params['nv']), 
>                                         np.radians(np.linspace(0, params['rotDirection']*(params['angle']-0.4), params['nProj'])),
>                                         (params['DSD']/(params['su']/params['nu'])), 
>                                         0)

Shifting the detector virtually into the volume works great. But for my understanding it should work also with no virtual shifting, just using the normal function. Additionally if I'm using a volume, which is smaller than the detector columns it just zoom in and don't reduce voxel size and also the alignment is going wrong after changing the reconstructions volume size. In my understand it should not have any influence of that. Correct me if I'm wrong. 

Can somebody help me with this? 

Thank you!

**Comment by meetaig** [comment]
Hi,
I saw you opened quite a lot of issues in here. Please note that this forum is for bugs and feature requests on the toolbox only and not usually for helping people use the toolbox. There are some help pages and presentation material available on the website (astra-toolbox.com) which hopefully should help you further. If you have additional questions feel free to contact the email given on the website.

However, in your case it should be a quick fix. You define the detector spacing as 0.24mm, but it does not do what you expect in the internals of the toolbox. ASTRA expects all numbers (distances, detector pixel size, etc.) to be in relation to the voxel size, which internally is set to 1. This means if you specify 0.24, you tell ASTRA that the detector pixels are a bit less than 1/4th of a voxel. This is probably not your intention. What you need to do is to take the pixel size 0.24mm and divide it by the size of the voxels you want to reconstruct. Let's say you have voxels of size 100 micron, you then divide 0.24 by 0.1 and that is what you need to give in as the detector spacing. Similarly you need to do this for the source origin and origin detector distances.

Hope this helps!

**Comment by MaxBe23** [comment]
Hey, 

@meetaig sorry for that, just wanted to help some people with their issues. Sorry, it was my bad. 

Back to the question. OK, I will givt it a trial. But then it should be fixed at the documentation, it is quite confusing for my understanding. 

It worked, thank you very much. 

In this case, it could be useful, to set the resulting voxel size as a parameter in the function: 

`astra.creators.create_vol_geom(x,y,z,voxel_size)`

To make it a bit more user friendly? 

Thank you again. 

Cheers

**Comment by meetaig** [comment]
Hi,
yea I only saw after I posted that you were actually helping people. This is very nice of you and greatly appreciated! We probably should have some kind of user forum for those kind of things, so the issue tracker does not get spammed with requests on using the toolbox.

Glad my tip helped you!

The model of ASTRA just includes the voxel size as a unit length, so it would be quite a lot of work to change it all I think. You're right about the parameter you mention, it might make some things easier. Like it is treated now it offers the same amount of flexibility as without it, however. Not having to deal with unit conversions does offer some advantages as well. So I dont think it will be changed anytime soon.
